src/GNNGraphs/transform.jl

```diff
@@ -38,6 +38,62 @@ function add_self_loops(g::GNNGraph{<:ADJMAT_T})
              g.ndata, g.edata, g.gdata)
 end
 
+"""
+    add_self_loops(g::GNNHeteroGraph, edge_t::EType)
+
+Return a graph with the same features as `g`
+but also adding self-loops of the specified type, edge_t
+
+Nodes with already existing self-loops of type edge_t will obtain a second self-loop of type edge_t.
+
+If the graphs has edge weights for edges of type edge_t, the new edges will have weight 1.
+"""
+function add_self_loops(g::GNNHeteroGraph{<:COO_T}, edge_t::EType)
+    function get_edge_weight_nullable(g::GNNHeteroGraph{<:COO_T}, edge_t::EType)
+        get(g.graph, edge_t, (nothing, nothing, nothing))[3]
+    end
+
+    src_t, _, tgt_t = edge_t
+    (src_t === tgt_t) ||
+        @error "cannot add a self-loop with different source and target types"
+    
+    n = get(g.num_nodes, src_t, 0)
+
+    # By avoiding using haskey, this only calls ht_keyindex once instead of twice
+    if (x = get(g.graph, edge_t, nothing)) !== nothing
+        s, t = x[1:2]
+        nodes = convert(typeof(s), [1:n;])
+        s = [s; nodes]
+        t = [t; nodes]
+    else
+        nodes = [1:n;]
```

Comment:
This creates a cpu array even if the rest of graph lives on gpu. We should create a new array similar to one in the existing relations.